image/tarexport/save.go

```diff
@@ -191,15 +189,13 @@ func (s *saveSession) save(outStream io.Writer) error {
 	manifestFileName := filepath.Join(tempDir, manifestFileName)
 	f, err := os.OpenFile(manifestFileName, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
 	if err != nil {
-		f.Close()
 		return err
 	}
+	defer f.Close()
```

Comment:
IMO this is fine. Unless we're doing some horrible magic with `system.Chtimes` that require us to first flush data before we change the modified time -- in that case the old way is less incorrect but would require us to do a `f.Close()` in every error path.